internal/encryption/encryption.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2024 Canonical Ltd.
+// SPDX-License-Identifier: AGPL-3.0
+
+package encryption
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/hex"
+	"fmt"
+	"io"
+
+	"github.com/canonical/identity-platform-admin-ui/internal/logging"
+)
+
+type Encrypt struct {
+	gcm cipher.AEAD
+
+	logger logging.LoggerInterface
+}
+
+// Encrypt takes a plain string and returns a hex encoded string
+func (e *Encrypt) Encrypt(data string) (string, error) {
+	payload := []byte(data)
+	nonce, err := e.generateCipherNonce()
+	if err != nil {
+		e.logger.Errorf("error generating random the nonce, %v", err)
+		return "", fmt.Errorf("error generating random the nonce, %v", err)
+	}
+	ciphertext := e.gcm.Seal(nonce, nonce, payload, nil)
+	return hex.EncodeToString(ciphertext), nil
+}
+
+// Decrypt takes hex encoded string and returns the decrypted plain string
+func (e *Encrypt) Decrypt(hexData string) (string, error) {
+	encrypted, err := hex.DecodeString(hexData)
+	if err != nil {
+		e.logger.Errorf("error decoding hex encoded string, %v", err)
+		return "", fmt.Errorf("error decoding hex encoded string, %v", err)
+	}
+
+	noncePart, payloadPart := encrypted[:e.gcm.NonceSize()], encrypted[e.gcm.NonceSize():]
+	decryptedData, err := e.gcm.Open(nil, noncePart, payloadPart, nil)
+	if err != nil {
+		e.logger.Errorf("error decrypting data: %v", err)
+		return "", fmt.Errorf("error decrypting data: %v", err)
+	}
+
+	return string(decryptedData), nil
+}
+
+var ioReadFull = io.ReadFull
```

Comment:
is this needed for anything more than the casting?if we want to keep it can we put it top of the page?

Reply:
we need this only for testing purposes as apparently it's the only accepted way of controlling nested functions during unit tests. I kept it close to where it's used only for clarity but I can move it up top no problem